### PR TITLE
intermediate commit

### PR DESCRIPTION
**Pull request by OttoVT**

### CODE REVIEW

File: src/VaultAgent.Common/Persistence/Transactions/TransactionSigningRequestEntity.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        }
+        public long Id { get; set; }
+        public long Sequence { get; set; }
+        public long VaultId { get; set; }
+        public string Component { get; set; }
+        public long OperationId { get; set; }
+        public string OperationType { get; set; }
+        public string BlockchainId { get; set; }
+        public string TenantId { get; set; }
+        public DateTime CreatedAt { get; set; }
```
Comment (KonstantinRyazantsev) [inline src/VaultAgent.Common/Persistence/Transactions/TransactionSigningRequestEntity.cs:21]: DateTimeOffset for EF

File: src/VaultAgent.Common/Persistence/Transactions/TransactionSigningRequestRepository.cs [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            try
+            {
+                await context.SaveChangesAsync();
+            }
+            catch (DbUpdateException e) when (e.InnerException is PostgresException pgEx && pgEx.SqlState == "23505")
+            {
+            }
+        }
+
+        public async Task AddAsync(TransactionSigningRequest transactionSigningRequest)
```
Comment (KonstantinRyazantsev) [inline src/VaultAgent.Common/Persistence/Transactions/TransactionSigningRequestRepository.cs:46]: Do we really need both of them?

File: src/VaultAgent/GrpcServices/TransactionsService.cs [hunk 3]
```diff
@@ -1,64 +1,187 @@
...
 
-                var transactionId = GetTransactionId(request.BlockchainId, signFacadeResponse.SignedTransaction);
-                
-                var response = new SignTransactionResponse
+                if (!outbox.IsStored)
                 {
-                    Response = new SignTransactionResponseBody
+                    var vault = await _vaultsRepository.GetByIdOrDefaultAsync(request.VaultId);
+
+                    if (vault == null)
```
Comment (KonstantinRyazantsev) [inline src/VaultAgent/GrpcServices/TransactionsService.cs:89]: shouldn't this be outside of the outbox? It's just a validation?

File: src/VaultAgent/GrpcServices/TransactionsService.cs [hunk 4]
```diff
@@ -1,64 +1,187 @@
...
+                                    null)
+                            }
+                        }
                     });
 
-                return response;
+                    _logger.LogInformation("Transaction Signing Request has been created {@context}",
+                        new
+                        {
+                            request.RequestId,
```
Comment (KonstantinRyazantsev) [inline src/VaultAgent/GrpcServices/TransactionsService.cs:169]: add the Id here

File: src/VaultAgent/GrpcServices/TransactionsService.cs (hunk 3 above)
Comment (OttoVT) [reply src/VaultAgent/GrpcServices/TransactionsService.cs:89]: Validation which includes database read operations should be inside outbox for performance optimization

File: src/VaultAgent.ApiContract/transactions.proto [hunk 5]
```diff
@@ -51,11 +51,9 @@ message TransactionSigningRequest {
 
   google.protobuf.StringValue rejectionReasonMessage = 13;
 
-  string transactionId = 14;
+  TransactionSigningRequestState state = 14;
 
-  TransactionSigningRequestState state = 15;
-
-  NullableTransactionSigningRequestRejectionReason rejectionReason = 16;
+  NullableTransactionSigningRequestRejectionReason rejectionReason = 15;
```
Comment (KonstantinRyazantsev) [inline src/VaultAgent.ApiContract/transactions.proto:56]: Avoid gaps in the fields numbers until we are in production. Fix it in the next task please

**Pull request merged**